Accessibility Fixes for Indigo Theme/Plugin Based on Audit

zubairshakoorarbisoft opened the pull request:
## Accessibility Fixes for Indigo Theme/Plugin

The audit team identified several accessibility-related issues within the Indigo Theme/Plugin. While investigating these issues, I discovered a package that required modifications to improve accessibility.

After a thorough review and debugging process, I implemented several fixes to address these problems. The issues were tracked on a Taiga board, and I’ve attempted to resolve all `brand-openedx` related issues within a single pull request.

### 🔧 Summary of Changes

- Enhanced focus visibility on the course title
- Discussion and Question icons on the Discussion page  
- Theme toggle button

I’ve included references to the corresponding issues below for additional context:

### Related Issues on Taiga

1. [TIA-53 – FOverlap between "Skip to Main Content" button and Menu Hamburger Icon at 200% Zoom. 2.4.11 Focus Not Obscured (Minimum)](https://tree.taiga.io/project/zaraahmed-tutor-indigo-accessibility/us/53)
2. [TIA-145 – Delete My Account" Button Lacks Focus Visibility](https://tree.taiga.io/project/zaraahmed-tutor-indigo-accessibility/us/145)
3. [TIA-58 – Course (Home) page: Icons under Course Tools fail the Non Text Contrast Requirement](https://tree.taiga.io/project/zaraahmed-tutor-indigo-accessibility/us/58)
4. [Discussion page: Non text contrast for discussion and question icon is below the minimum requirment](https://tree.taiga.io/project/zaraahmed-tutor-indigo-accessibility/us/45)
5. [Admin & Student Profile Page: Theme toggle button fails contrast requirements](https://tree.taiga.io/project/zaraahmed-tutor-indigo-accessibility/us/83)


---

### 🖼️ Screenshots/Sandbox


| Before | After |
|--------|-------|
| ![image](https://github.com/user-attachments/assets/84cf77a1-6579-4dc7-b138-702354a20a43) |  ![image](https://github.com/user-attachments/assets/1afc32e4-a34c-4d7e-9c74-8040ce177ac1) | 

---

| Before | After |
|--------|-------|
|  ![image](https://github.com/user-attachments/assets/60ef7424-b2a5-4571-86cc-9253e52302b1) | ![image](https://github.com/user-attachments/assets/c9a18d78-43bb-4340-a64b-d88aa60c8fab) |

---

| Before | After |
|--------|-------|
| ![image](https://github.com/user-attachments/assets/7f0a0def-4ad4-4478-93a9-33ed99c6c1fa) | ![image](https://github.com/user-attachments/assets/fe1aab1d-93b8-4a45-afb9-bb6c1d88ae62) |


---

| Before | After |
|--------|-------|
| ![image](https://github.com/user-attachments/assets/65ee6913-4fa0-4d25-b633-b242f4cb30e1) |  ![image](https://github.com/user-attachments/assets/0217eb36-1a9e-45c4-b721-397ed46b91af) |


---

| Before | After |
|--------|-------|
|  ![image](https://github.com/user-attachments/assets/43f2b63a-ebae-4ff8-8b8a-a85fa986bced) |  ![image](https://github.com/user-attachments/assets/2d1b551f-b587-4d28-9acb-b658d1fe57c5) |



Please refer to the linked issues above for a deeper understanding of the context and the improvements made.
Related to https://github.com/overhangio/tutor-indigo/issues/146

Commit:
fix: foucus visible overriden issue fixed TAI-28

Commit:
fix: buttons focus styling fixes added

Commit:
fix: removed extra clesses from styl

Commit:
fix: TIA-53: Skip to mian content overlap issue fixed

Commit:
fix: TIA-76 Fixes focus styling for account screen

Commit:
fix: TIA-87 fixed focus issue for explore courses and Submit button

Commit:
fix: TIA-145 Fixed focus visibility issue for Delete my account button

Commit:
fix: course title focus styling fixed

ahmed-arb commented on paragon/_account.scss:
```diff
@@ -259,3 +259,14 @@
     } 
 }
+// button focus styling
+.btn-brand,
+.btn-primary,
+.btn-tertiary,
+.btn-outline-danger,
+.btn-outline-primary,
+.btn-inverse-outline-primary {
+    &:focus {
+        box-shadow: 0 0 0 2px #015fcc !important;
+    }
+}
```
I am not sure, but general changes like these should be in generic files like `_base.scss` or `_overrides.scss`. So that we don't repeat them over and over, like we are doing in this PR.

zubairshakoorarbisoft replied:
@ahmed-arb thanks for the review. I am going to remove this piece of code as rahat has made a generic fix in [this PR](https://github.com/edly-io/brand-openedx/pull/36) which is also fixing this problem as well but another way.

arbirali replied:
This code has been removed.

ahmed-arb commented on paragon/_header.scss:
There's a github "No newline at the end of file" error here.

zubairshakoorarbisoft replied:
done

ahmed-arb reviewed (commented):
Thanks for this PR! 

To help with the review, could you please:

* **Provide "before and after" screenshots or a short video** for the fixes/changes addressed in this PR.
* **Summarize the specific changes made** in this PR e.g "Improved button focus styling", "Adjusted 'Skip to Main Content' button positioning", etc.

This will greatly assist in verifying the fixes and ensure everything is working as expected.

Commit:
fix: TIA-58 updated course tools icon color according WCAG

Commit:
fix: removed redundant changes

Commit:
fix: TIA-83 update color contrast for theme toggle button

Commit:
fix: new line added in _header.scss bottom

zubairshakoorarbisoft commented:
> Thanks for this PR!
> 
> To help with the review, could you please:
> 
> * **Provide "before and after" screenshots or a short video** for the fixes/changes addressed in this PR.
> * **Summarize the specific changes made** in this PR e.g "Improved button focus styling", "Adjusted 'Skip to Main Content' button positioning", etc.
> 
> This will greatly assist in verifying the fixes and ensure everything is working as expected.

@ahmed-arb , there are a lot of tickets addressed int this PR, so attaching screenshots for each cause PR messy, that's why attached taiga tickets links in the description, they have pretty details of the issue been addressed.

Commit:
fix: TIA-45 updated discussions and questions color according WCAG standard

zubairshakoorarbisoft committed:
Merge branch 'sumac/indigo' into zubair/tutor-indigo-fixes

arbirali committed:
fix: feedback implementations

arbirali commented:
@ahmed-arb  I’ve updated the PR description to include a summary of the specific changes made and also added before/after screenshots for clarity. Let me know if there’s anything else you'd like me to adjust.

CC: @zubairshakoorarbisoft

zubairshakoorarbisoft commented:
> @ahmed-arb I’ve updated the PR description to include a summary of the specific changes made and also added before/after screenshots for clarity. Let me know if there’s anything else you'd like me to adjust.
> 
> CC: @zubairshakoorarbisoft

@ahmed-arb schreenshot has been attached, please have a final look into it again.

Thanks @arbirali
